packages/cli/src/commands/generate/function/function.js

```diff
@@ -54,13 +54,53 @@ export const builder = (yargs) => {
         'https://redwoodjs.com/reference/command-line-interface#generate-function'
       )}`
     )
+
   // Add default options, includes '--typescript', '--javascript', '--force', ...
   Object.entries(yargsDefaults).forEach(([option, config]) => {
     yargs.option(option, config)
   })
 }
 
-export const { command, handler } = createYargsForComponentGeneration({
-  componentName: 'function',
-  filesFn: files,
-})
+// This could be built using createYargsForComponentGeneration;
+// however, we need to add a message after generating the function files
+export const handler = async ({ name, force }) => {
```

Comment:
This needs to take the --javascript and --typescript options, defaulting to javascript.

Currently, the command _only_ generates .ts regardless

Reply:
Higher-level question not to be resolved in this PR --> for our generator commands in general, do we have unit tests confirming the conversion to JS is working? AND that TS files can be generated as well. 🤔

Reply:
Oh - when I read the comment in 

```js
  // Add default options, includes '--typescript', '--javascript', '--force', ...
  Object.entries(yargsDefaults).forEach(([option, config]) => {
    yargs.option(option, config)
  })
```
I assumed it did that -- but apparently it doesn't. Well it did add --force so I imagine the --javascript is added too but just not handled.

Reply:
> for our generator commands in general, do we have unit tests confirming the conversion to JS is working?

Yes, from a templating standpoint:

```js
  javascriptFiles = functionGenerator.files({
    name: 'javascript-function',
    javascript: true,
  })
  typescriptFiles = functionGenerator.files({
    name: 'typescript-function',
    typescript: true,
  })
```

The issue here is that it sets the `javascript` or `typescript` but in the function generator, that param wasn't being passed to the `files()`.

Reply:
> Currently, the command _only_ generates .ts regardless

Fixed:

```terminal

rw_tests % yarn rw g function myTypescriptFunction --typescript 
yarn run v1.22.10
$ '/rw_tests/node_modules/.bin/rw' g function myTypescriptFunction --typescript
  ✔ Generating function files...
    ✔ Successfully wrote file `./api/src/functions/myTypescriptFunction.ts`

⚠ Important:

When deployed, a custom serverless function is an open API endpoint and is your responsibility to secure appropriately.

Please consult the Serverless Function Considerations (​https://redwoodjs.com/docs/serverless-functions#security-considerations​) in the RedwoodJS documentation for more information.

✨  Done in 2.10s.
rw_tests % yarn rw g function myJavascriptFunction --javascript 
yarn run v1.22.10
$ '/rw_tests/node_modules/.bin/rw' g function myJavascriptFunction --javascript
  ✔ Generating function files...
    ✔ Successfully wrote file `./api/src/functions/myJavascriptFunction.js`

⚠ Important:

When deployed, a custom serverless function is an open API endpoint and is your responsibility to secure appropriately.

Please consult the Serverless Function Considerations (​https://redwoodjs.com/docs/serverless-functions#security-considerations​) in the RedwoodJS documentation for more information.

✨  Done in 2.41s.
rw_tests % yarn rw g function myDefaultJavascriptFunction              
yarn run v1.22.10
$ '/rw_tests/node_modules/.bin/rw' g function myDefaultJavascriptFunction
  ✔ Generating function files...
    ✔ Successfully wrote file `./api/src/functions/myDefaultJavascriptFunction.js`

⚠ Important:

When deployed, a custom serverless function is an open API endpoint and is your responsibility to secure appropriately.

Please consult the Serverless Function Considerations (​https://redwoodjs.com/docs/serverless-functions#security-considerations​) in the RedwoodJS documentation for more information.

✨  Done in 2.30s.
```